config/v1/service.proto

```
@@ -0,0 +1,292 @@
+  // for several watches at once. 
+  rpc Watch(stream WatchRequest) returns (stream WatchResponse) {
+      option (google.api.http) = {
+        post: "/events/v1:watch"
+        body: "*"
+    };
+  }
+}
+
+message ObjectFieldInclude {
+  // if true object(s) should include data.
+  bool data = 1;
+
+  // if true object(s) should include source_data.
+  bool source_data = 2;
+}
+
+message ListObjectTypesRequest {
+  // meta is the metadata associated with the root where listing begins.
+  Meta meta = 1;
+}
+
+message ListObjectsRequest {
+  // meta is the metadata associated with the root where listing begins.
+  Meta meta = 1;
+
+  // select selects fields to include in the result.
+  ObjectFieldInclude incl = 2; 
+}
+
+message UpdateObjectRequest {
+  // meta is the metadata associated with the object.
+  Meta meta = 1;
+  
+  // source_data is the data as it was specified in json / yaml format.
+  google.protobuf.Struct source_data = 2;
+}
+
+message CreateObjectRequest {
+  // meta is the metadata associated with the object.
+  Meta meta = 1;
+  
+  // source_data is the data as it was specified in json / yaml format.
+  google.protobuf.Struct source_data = 2;
+}
+
+message GetObjectRequest {
+  // meta is the metadata associated with the object.
+  Meta meta = 1;
+
+  // incl(ude) selects fields to include in the result.
+  ObjectFieldInclude incl = 2; 
+}
+
+message DeleteObjectRequest {
+  // meta is the metadata associated with the object.
+  Meta meta = 1;
+}
+
+// Object
+message Object {
+  // meta is the metadata associated with the object.
+  Meta meta = 1;
+
+  // source_data is the data as it was specified in json / yaml format.
+  google.protobuf.Struct source_data = 2;
+
+  // data is the binary encoded protobuf data
+  // Both representations are useful. 
+  // For example:
+  // If `souce_data` contains an expressions in text form 
+  // `data` may contain a parsed AST
+  bytes data = 3; 
```

it fails hard, if the underlying protos are not available. This establishes an unwanted coupling.